includes/class-wc-cart.php

```diff
@@ -1354,6 +1357,13 @@ public function get_shipping_packages() {
 				),
 			)
 		);
+
+		// Add an index/ID to each package in the array using the key.
+		foreach ( $packages as $key => $package ) {
+			$packages[ $key ]['package_id'] = $key;
```

Comment:
probably call this `package_index` to indicate that this should not be used as a unique key.
Or, add a check for whether `package_id` is already present, incase an extension is adding it, for eg, to identify different boxes.

Reply:
The usage we found did the same thing - numeric index from key.